builtin/logical/postgresql/secret_creds.go

```diff
@@ -91,12 +91,25 @@ func (b *backend) secretCredsRevoke(
 	}
 	username, ok := usernameRaw.(string)
 
+
 	// Get our connection
 	db, err := b.DB(req.Storage)
 	if err != nil {
 		return nil, err
 	}
 
+	// Check if the role exists
+	var exists bool
+	query := fmt.Sprintf("SELECT exists (SELECT rolname FROM pg_roles WHERE rolname='%s');", username);
```

Comment:
Mandatory change: using `fmt.Sprintf` to generate SQL is not permitted and must be changed to use `db.Query()` or `db.QueryRow()` otherwise there's nothing preventing a visit from Bobby Tables:

```
db.QueryRow(`SELECT exists (SELECT rolname FROM pg_roles WHERE rolname='$1');", username)
```